pkg/auth/auth.go

```diff
@@ -114,6 +117,23 @@ func NewAuthorizerForAuth(tokenString string, certificate string, server string)
 	return NewBearerAuthorizer(JwtTokenProvider{tokenString}, transportCreds), nil
 }
 
+func NewAuthorizerForAuthFromCACertHash(tokenString string, cacerthash string, server string) (Authorizer, error) {
+	pkv := NewPublicKeyVerifier()
+	err := pkv.Allow(cacerthash)
+	if err != nil {
+		return NewBearerAuthorizer(JwtTokenProvider{}, credentials.NewTLS(nil)), fmt.Errorf("could not marshal the server certificate")
+	}
+
+	transportCreds := credentials.NewTLS(&tls.Config{
+		ServerName:            server,
+		InsecureSkipVerify:    true,
+		VerifyPeerCertificate: pkv.VerifyPeerCertificate,
```

Comment:
Not related to PR; should we look into using VerifyConnection instead?

Reply:
lets discuss this further.